cluster/src/dunit/scala/io/snappydata/cluster/SecureQueryRoutingDUnitTest.scala

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{Connection, DriverManager, ResultSet}
+
+import com.pivotal.gemfirexd.Attribute
+import com.pivotal.gemfirexd.security.{LdapTestServer, SecurityTestUtils}
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+class SecureQueryRoutingDUnitTest(val s: String)
+    extends ClusterManagerTestBase(s) with Logging {
+
+  val snappyAdminUser = "gemfire2"
```

Comment:
This query routing dunit file is not part of this PR. It got in here due to a merge went awry. Will update.